common/src/main/java/org/apache/carbondata/common/exceptions/sql/NoSuchCarbonTableException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.common.exceptions.sql;
+
+import org.apache.carbondata.common.annotations.InterfaceAudience;
+import org.apache.carbondata.common.annotations.InterfaceStability;
+
+@InterfaceAudience.User
+@InterfaceStability.Stable
+public class NoSuchCarbonTableException extends MalformedCarbonCommandException {
+  public NoSuchCarbonTableException(String db, String table) {
```

Comment:
I feel this PR may not be required
1. Spark already provides `NoSuchTableException`. We can use the same to throw the error if required
2. You are making the use of this exception in DDLStrategy. In DDLstrategy we specifically check for CarbonTable. If CarbonTable is not found then it can also be a parquet or hive table. So I think either we should continue with current malformed exception being thrown or we should pass the flow to hive/parquet and let the exception be thrown from there.

Reply:
@manishgupta88

1.  i think MalformedCarbonCommandException is not right, user will not know the real exception
2.  the ddl is used for carbon, so i don't think we should pass the flow to hive
3.  Do you think which is better?  NoSuchTableException or NoSuchCarbonTableException , or some better suggests?

Reply:
@qiuchenjian ..
1. Carbon in one of the component integrated with spark just like hive and we can multiple components integrated together in one system with spark. In `DDLStratgey` we are just tapping the commands to check for the carbon table. If its a carbon table we give to carbon flow else we give it to spark. We are throwing Malformed exception because spark itself does not support the Alter operation.
2. Still If you want to change the Malformed exception you can change it to `NoSuchTableExcpetion`. Its a generic exception and should be sufficient.

Reply:
@manishgupta88  I still suggest to mention user more clearly,  i change it to `NoSuchTableException`,  please review again, thank you

Reply:
@manishgupta88  discuss with david (@QiangCai ), he suggest to use MalformedCarbonCommandException and modify the message to "Table or view 'no_table' not found in database 'default' or not carbon table"

Reply:
@qiuchenjian ...This was the very first comment that we throw `MalformedCarbonCommandException` for a reason and the PR may not be required but you were not convinced. Now that you have taken another opinion you can make the required changes and get it reviewed and merged